tests/periph_uart/main.c

```diff
@@ -154,6 +235,32 @@ static int cmd_init(int argc, char **argv)
     return 0;
 }
 
+#ifdef MODULE_PERIPH_UART_MODECFG
+static int cmd_mode(int argc, char **argv)
+{
+    int dev, res;
+
+    if (argc < 5) {
+        printf("usage: %s <dev> <data bits> <parity> <stop bits>\n", argv[0]);
+        return 1;
+    }
+
+    dev = parse_dev(argv[1]);
+    if (dev < 0) {
+        return 1;
+    }
+    res = uart_mode(dev, parse_data_bits(argv[2]), parse_parity(argv[3]),
```

Comment:
this does potentiall break pretty nicely :-) 

-> **must** use `UART_DEV(dev)` as first parameter

Reply:
Reworked to check return values of parse_data_bits&Co before calling uart_mode